### PR TITLE
Remove `ballast_physical_*_level`

### PR DESCRIPTION
**Pull request by kevin-david**
This is not settable, and the range is 0-255 not 1-254

```
Read result of 'lightingBallastCfg': {"physicalMinLevel":0,"physicalMaxLevel":255}
```

and
```
First occurred: October 16, 2022 at 11:22:54 PM (2670 occurrences)
Last logged: 2:44:12 PM

    Invalid value for number.0x25c823ffffa60700_ballast_physical_minimum_level: 0 (range 1.0 - 254.0)
    Invalid value for number.0x25c823ffffa60700_ballast_physical_maximum_level: 255 (range 1.0 - 254.0)
```

and

```
Publish 'set' 'write' to '0x25c823ffffa60700' failed: 'Error: Write 0x25c823ffffa60700/1 lightingBallastCfg({"physicalMinLevel":6}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'READ_ONLY')'
```

### CODE REVIEW

File: devices/leviton.js [hunk 1]
```diff
@@ -101,9 +101,9 @@ module.exports = [
                 .withDescription('Specifies the maximum brightness value'),
             exposes.numeric('ballast_power_on_level', ea.ALL).withValueMin(1).withValueMax(254)
                 .withDescription('Specifies the initialisation light level. Can not be set lower than "ballast_minimum_level"'),
-            exposes.numeric('ballast_physical_minimum_level', ea.ALL).withValueMin(1).withValueMax(254)
+            exposes.numeric('ballast_physical_minimum_level', ea.STATE_SET).withValueMin(0).withValueMax(255)
```
Comment (kevin-david) [inline devices/leviton.js:104]: @Koenkk what's the right way to make this actually read only? 

is it removing https://github.com/Koenkk/zigbee-herdsman-converters/blob/2103a26beb34e2dff1f11afc29e16d37b377db8f/converters/toZigbee.js#L581? 

I think this attribute _might_ be broadly read only, it definitely is for this model, but I haven't been able to dig up a lot of documents...

Comment (kevin-david) [reply devices/leviton.js:104]: Hm, that didn't seem to work (https://github.com/Koenkk/zigbee-herdsman-converters/pull/4813/commits/a5561a910fc382c8834565396108ad34dabb5be5)

Comment (Koenkk) [reply devices/leviton.js:104]: Read only = `ea.STATE_GET` 

- So the value can be found in the published state
- And you can do a `/get`

Comment (kevin-david) [reply devices/leviton.js]: Whoops, I did `STATE_SET` by accident on the first one. Updated now.

Either way, how do I get the CI to pass now - is there something else I have to change? https://github.com/Koenkk/zigbee-herdsman-converters/actions/runs/3303623754/jobs/5451785989

Thanks!

**Comment by Koenkk** [comment]
@sjorge do I remember correctly that you have a Ubisys D1? If yes, do you know if the `ballast_physical_minimum_level` is settable?

**Comment by sjorge** [comment]
> @sjorge do I remember correctly that you have a Ubisys D1? If yes, do you know if the `ballast_physical_minimum_level` is settable?

The D1 is the only one I don’t have. I got a bunch of S1, S2, J1’s and one C4.

**Comment by sjorge** [comment]
@Koenkk that being said, I had a look at the ZCL as this is just a standard cluster.

image

It looks like those are indeed read only, they are similar to the colorTemperature min/max stuff. It just describes what the devices actually capabilities are. I don't think it is very useful to have those readable as an attribute in the mqtt payload.

I think it would make sense to treat them like the colorTemperature min/max and use there values in the exposes information to correctly set the range.

**Comment by Koenkk** [comment]
> It looks like those are indeed read only, they are similar to the colorTemperature min/max stuff. It just describes what the devices actually capabilities are. I don't think it is very useful to have those readable as an attribute in the mqtt payload.

I agree with this. The people using this attribute is probably very limited and they can always read it via the dev console.

@kevin-david could you remove the `ballast_physical_*_level` completely (so also the exposes) for all devices.

**Comment by kevin-david** [comment]
@Koenkk 
> > It looks like those are indeed read only, they are similar to the colorTemperature min/max stuff. It just describes what the devices actually capabilities are. I don't think it is very useful to have those readable as an attribute in the mqtt payload.
> 
> I agree with this. The people using this attribute is probably very limited and they can always read it via the dev console.
> 
> @kevin-david could you remove the `ballast_physical_*_level` completely (so also the exposes) for all devices.

Done - I agree as well, the property isn't useful generally.

File: devices/leviton.js (hunk 1 above)
Comment (kevin-david) [reply devices/leviton.js]: Obsolete with direction https://github.com/Koenkk/zigbee-herdsman-converters/pull/4813#issuecomment-1288036529

**Comment by kevin-david** [comment]
@Koenkk snuck one more commit in here, I missed `fz.on_off` when adding this device originally back in #4697 

I think this will fix the error I have locally: `No converter available for 'ZS057-D0Z' with cluster 'genOnOff' and type 'attributeReport' and data '{"onOff":0}'`

BTW, is there any easy way to test PRs via home assistant  other than setting up a custom converter temporarily? I saw https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html#home-assistant-addon which only applies to the dev branch and I assume there's some kind of publishing step...

**Comment by Koenkk** [comment]
@kevin-david to test locally, get root access to your HA container, enter the z2m container by doing `docker exec -it Z2M_CONTAINER_ID /bin/bash` , then the zigbee-herdsman-converters files can be found under `node_modules/zigbee-herdsman-converters`

**Comment by kevin-david** [comment]
@Koenkk 
> @kevin-david to test locally, get root access to your HA container, enter the z2m container by doing `docker exec -it Z2M_CONTAINER_ID /bin/bash` , then the zigbee-herdsman-converters files can be found under `node_modules/zigbee-herdsman-converters`

The changes seem to get stomped on when I restart the addon from HA, and restarting via the web frontend doesn't seem to have any effect. Doing a `kill -HUP` on the `node index.js` process also seems to kill and restart the container, which drops the changes as well.

I'm editing the path `/app/node_modules/zigbee-herdsman-converters` with `vi` while the container is running - does that sound right?

Thanks again!

**Comment by Koenkk** [comment]
You should not restart via HA itself but via `docker restart CONTAINER_ID`. Restarting via HA will delete the container.

**Comment by kevin-david** [comment]
> You should not restart via HA itself but via `docker restart CONTAINER_ID`. Restarting via HA will delete the container.

Thanks, that helped. The other thing I had to do was turn off the watchdog functionality on the addon so it wasn't automatically restarted. I think this is worth documenting, any suggestions on where?

Also, I broke out the Leviton specific changes into #4834 since they were unrelated

**Comment by Koenkk** [comment]
@kevin-david thanks, I believe the best place is here: https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/blob/master/README.md

**Comment by kevin-david** [comment]
@Koenkk 

> @kevin-david thanks, I believe the best place is here: https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/blob/master/README.md

Here's a PR: https://github.com/zigbee2mqtt/hassio-zigbee2mqtt/pull/406

**Pull request merged**